src/components/hoc/withAdapter.test.jsx

```diff
@@ -0,0 +1,71 @@
+import React, {useContext} from 'react';
+import {create, act} from 'react-test-renderer';
+import waitForExpect from 'wait-for-expect';
+import PropTypes from 'prop-types';
+
+import withAdapter from './withAdapter';
+import WebexJSONAdapter from '../../adapters/WebexJSONAdapter';
+import {AdapterContext} from '../hooks';
+
+describe('withAdapter HOC', () => {
+  const adapterConnectedChanged = jest.fn();
+  const contextAdapterChanged = jest.fn();
+  const customPropsChanged = jest.fn();
+  let adapter;
+  let testRenderer;
+
+  function MyComponent({adapterConnected, prop1, prop2}) {
+    adapterConnectedChanged(adapterConnected);
+    contextAdapterChanged(useContext(AdapterContext));
+    customPropsChanged({prop1, prop2});
+
+    return <div />;
+  }
+
+  MyComponent.propTypes = {
+    adapterConnected: PropTypes.bool.isRequired,
+    prop1: PropTypes.string.isRequired,
+    prop2: PropTypes.string.isRequired,
+  };
+
+  const MyComponentWithAdapter = withAdapter(MyComponent, () => {
+    adapter = new WebexJSONAdapter({});
+    adapter.connect = jest.fn(adapter.connect);
+    adapter.disconnect = jest.fn(adapter.disconnect);
+
+    return adapter;
+  });
+
+  act(() => {
```

Comment:
regarding the `act()` error, the offending code is outside of our reach (part of `withAdapter`'s definition), so we have nothing else to wrap in `act()`. I suspect some feature for testing hooks is missing in react-test-renderer.

Reply:
@cipak, try wrapping the lines in `MyComponentWithAdapter` inside the `act()`. Creating the wrapped component is part of the test setup

Reply:
@lalli-flores I tried it and it still gives me the warning.